src/lib/core/bidi/directionality.ts

```diff
@@ -0,0 +1,43 @@
+import {
+  EventEmitter,
+  Injectable,
+  Optional,
+  SkipSelf
+} from '@angular/core';
+
+export type Direction = 'ltr' | 'rtl';
+
+/**
+ * The directionality (LTR / RTL) context for the application (or a subtree of it).
+ * Exposes the current direction and a stream of direction changes.
+ */
+@Injectable()
+export class Directionality {
+  value: Direction = 'ltr';
+  public change = new EventEmitter<void>();
```

Comment:
Need unit tests for the following (creating a new `bidi.spec.ts`):
- [ ] Directionality reads `dir` from the html element if not specified on the body
- [ ] Directionality reads `dir` from the body even it is also specified on the html element
- [ ] Directionality defaults to `ltr` if nothing is specified on either body or the html element
- [ ] `Dir` provides itself as `Directionality` 
- [ ] `Dir` emits a change event when the value changes